### ISSUE TITLE
many drivers vs many cars with complex constraints possible ?

### ISSUE THREAD

**Issue by gustav3d** [reinterpretcat/vrp#58]
Hello,

I want to tank you for doing this library in Rust  :)

I wonder if its possible to modell with pickup deliveries fixed time windows, fixed breaks and exclusion((Some drivers can only use certain cars due to several criteria like drivers license, and cant be in same Car as certain gods(example:people that hate each other, or is allergic to dog))), certain goods cant be combined either -  constraints for many Drivers vs many Cars?.

Thanks
  Gustav
.

**Comment by reinterpretcat** [comment]
Hi,

the solver doesn't support drivers with pragmatic format yet, only vehicle types are supported. 

> I wonder if its possible to modell with pickup deliveries fixed time windows, fixed breaks 

These features are supported. `Exclusion` I guess can be modeled with skills feature. Relevant documentation:


https://reinterpretcat.github.io/vrp/concepts/pragmatic/problem/jobs.html#pickup-and-delivery-job
https://reinterpretcat.github.io/vrp/examples/pragmatic/basics/skills.html

**Comment by gustav3d** [comment]
is drivers supported if coding directly to the api ?

**Comment by reinterpretcat** [comment]
No, it is not. However, the internal api is designed to support this in the future:

https://github.com/reinterpretcat/vrp/blob/master/vrp-core/src/models/problem/fleet.rs#L30-L44

Check its usage

**Comment by gustav3d** [comment]
Aha !,
Thanks a lot for such a quick response.

**Comment by gustav3d** [comment]
Google Or-tools does not support drivers either, but there i fake it with cars. 
I dont think thats optimal though.

**Comment by reinterpretcat** [comment]
My idea was to use `Actor` model within tour which consists of joined data from a driver and a car:

https://github.com/reinterpretcat/vrp/blob/master/vrp-core/src/models/problem/fleet.rs#L95-L104

At the moment, the missing part is to build actors from multiple drivers and vehicles. The current implementation assumes that there is only one driver:

https://github.com/reinterpretcat/vrp/blob/master/vrp-core/src/models/problem/fleet.rs#L130-L131

**Comment by gustav3d** [comment]
Any discord chat for this project ?.

**Comment by reinterpretcat** [comment]
No, not yet. Maybe `issues` and `discussions` serves better from documentation point of view? Chat is good as it is nearly real time communication channel, but it is harder to seek in it for information later. So far, I would recommend to create a new discussion for questions and a new issue for bugs

**Comment by gustav3d** [comment]
Discord has full history search by default.
